storybookjs/storybook
# Build: Add benchmark task

Closes https://github.com/storybookjs/storybook/issues/21511

## What I did

I replaced our script for bench-marking storybook.
The `@storybook/addon-bench` is now deprecated, no longer used.
I've archived that repository.

I'm now tracking a bunch of more meaningful data about storybook:
https://github.com/storybookjs/storybook/blob/ac6984f038b896c3acd5b253c5f82a6a7265a852/scripts/bench/types.ts#L3-L61

This should make it easier for us to detect regressions in performance and size, and also make it easier to verify if what we're doing has impact we think it should have.

## How to test

When running sandboxes, a `bench.json` file should be created when tasks run.
Each task is responsible for tracking it's own metrics. This is to ensure we can expand this system over time.

I did make 1 `bench` task that collects data via playwright.

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
